lib/ansible/modules/cloud/amazon/ec2_vol_facts.py

```diff
@@ -59,57 +60,65 @@
 import traceback
 
 try:
-    import boto.ec2
-    from boto.exception import BotoServerError
-    HAS_BOTO = True
+    from botocore.exceptions import ClientError
+    from botocore.exceptions import ProfileNotFound
 except ImportError:
-    HAS_BOTO = False
+    pass  # caught by imported HAS_BOTO3
 
 from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.ec2 import connect_to_aws, ec2_argument_spec, get_aws_connection_info
+from ansible.module_utils.ec2 import connect_to_aws, ec2_argument_spec, get_aws_connection_info, boto3_conn, HAS_BOTO3, boto3_tag_list_to_ansible_dict
+from ansible.module_utils.ec2 import ansible_dict_to_boto3_filter_list, camel_dict_to_snake_dict
 from ansible.module_utils._text import to_native
 
 
-def get_volume_info(volume):
+def get_volume_info(volume, region):
 
-    attachment = volume.attach_data
+    attachment = volume["attachments"]
+    tags = {}
+    for tag in volume["tags"]:
```

Comment:
You've import boto3_tag_list_to_ansible_dict so you could use that.
line 99: `'tags': boto3_tag_list_to_ansible_dict(volume['tags'])`